app/models/animal.rb

```diff
@@ -17,49 +17,31 @@ class Animal < ApplicationRecord
             presence: true
 
   def predators_with_data
-    predators = self.predators.gsub(/\band\b/, '').split(',').map(&:strip).map(&:singularize)
-    predators_data = predators.flat_map do |predator_name|
-      Animal.where("name ILIKE ?", "%#{predator_name}%")
-    end
-    predators_data
+    predator_names = self.predators.gsub(/\band\b/, '').split(',').map(&:strip).map(&:singularize)
+    Animal.where('name ILIKE ANY (ARRAY[?])', predator_names.map { |name| "%#{name}%" })
   end
 
   def self.find_animal(params)
-
-    where("name ILIKE ?", "%#{params[:name]}%")
+    find_by("name ILIKE ?", "%#{params[:name]}%")
   end
 
   def self.handle_predator_creation(animal)
-    predators = Animal.format_predators(animal)
-    predators_data = predators.each_with_object([]) do |predator_name, created_preds|
-      # each w obj allows us to array up the preds and return at once.  
-      # in case pred_name fails
-      unless Animal.where("name ILIKE ?", predator_name).exists?
-
-        animal_response = AnimalGateway.fetch_animal_data(predator_name)
-        photo_response = AnimalGateway.fetch_photo_data(predator_name)
-        if animal_response && photo_response
-          new_animal = AnimalDetail.new(animal_response, photo_response).as_json
-          created_preds << Animal.create(new_animal)
-        end
+    predators = format_predators(animal)
+    predators.each do |predator_name|
```

Comment:
Good ole' .each.  The solution to the this problem has been 1.5 weeks in the making!